app/trip.py

```diff
@@ -0,0 +1,64 @@
+from datetime import datetime
+from math import dist
+from app.customer import Customer
+from app.shop import Shop
+
+
+def trip(
+        customer: Customer,
+        shops: list[Shop], fuel_price: float
+) -> None:
+    print(f"{customer.name} has {customer.money} dollars")
+
+    min_price = float("inf")
+    min_shop = None
+
+    for shop in shops:
+        distance = dist(customer.location, shop.location)
+        fuel_consumption = customer.car.fuel_consumption
+        distance_price = (distance / 100) * fuel_consumption * fuel_price * 2
+        product_price = 0
+
+        for product, amount in customer.product_cart.items():
+            cost = shop.products[product] * amount
+            product_price += cost
+
+        if distance_price + product_price < min_price:
+            min_price = distance_price + product_price
+            min_shop = shop
```

Comment:
Separate all calculation logic into different functions . That will be more clean imho